src/access/byTenant.ts

```diff
@@ -37,7 +37,7 @@ export const byTenant: (method: ruleMethod, collection: ruleCollection) => Acces
       )
       .map((assignment) => assignment.tenant)
       .filter((tenant) => typeof tenant !== 'number') // captured in the getter
-      .map((tenant) => tenant.id)
+      .map((tenant) => tenant?.id)
```

Comment:
Can we improve the `filter` call so `matchingTenantIds` doesn't end up as `[](number | undefined)`?